lib/join/server_token.go

```diff
@@ -39,8 +39,11 @@ func (s *Server) handleTokenJoin(
 	// Set any diagnostic info from the ClientParams.
 	setDiagnosticClientParams(stream.Diagnostic(), &tokenInit.ClientParams)
 
-	// There are no additional checks for the token join method, just make the
-	// result message and return it.
+	// make sure the secret provided in the TokenInit matches the token's secret
+	if tokenInit.Secret != token.GetSecret() {
```

Comment:
tbh I don't love how this obfuscates the fact that, for non-scoped token joins, the client sent no secret and the token has no secret and nothing is really checked here

Reply:
Yeah I like this better :+1: